lib/src/native/realm_core.dart

```diff
@@ -148,40 +147,176 @@ class _RealmCore {
+        if (config.initialDataCallback != null) {
+          _realmLib.realm_config_set_data_initialization_function(
+              configHandle._pointer, Pointer.fromFunction(initial_data_callback, FALSE), config.toWeakHandle());
+        }
+        if (config.isReadOnly) {
+          _realmLib.realm_config_set_schema_mode(configHandle._pointer, realm_schema_mode.RLM_SCHEMA_MODE_IMMUTABLE);
+        }
+        if (config.isInMemory) {
+          // TODO: Get rid of this
+          _realmLib.realm_config_set_in_memory(configHandle._pointer, config.isInMemory);
+        }
+        if (config.fifoFilesFallbackPath != null) {
+          _realmLib.realm_config_set_fifo_path(configHandle._pointer, config.fifoFilesFallbackPath!.toUtf8Ptr(arena));
+        }
+        if (config.disableFormatUpgrade) {
+          _realmLib.realm_config_set_disable_format_upgrade(configHandle._pointer, config.disableFormatUpgrade);
+        }
+        if (config.shouldCompactCallback != null) {
+          _realmLib.realm_config_set_should_compact_on_launch_function(
+              configHandle._pointer, Pointer.fromFunction(should_compact_callback, 0), config.toWeakHandle());
+        }
+      } else if (config is InMemoryConfiguration) {
+        _realmLib.realm_config_set_in_memory(configHandle._pointer, true);
+      } else if (config is FlexibleSyncConfiguration) {
+        final syncConfigPtr = _realmLib.invokeGetPointer(() => _realmLib.realm_flx_sync_config_new(config.user.handle._pointer));
+        try {
+          _realmLib.realm_sync_config_set_session_stop_policy(syncConfigPtr, config.sessionStopPolicy.index);
+          _realmLib.realm_config_set_sync_config(configPtr, syncConfigPtr);
+        } finally {
+          _realmLib.realm_release(syncConfigPtr.cast());
+        }
       }
 
       return configHandle;
     });
   }
 
+  SubscriptionSetHandle getSubscriptions(Realm realm) {
+    return SubscriptionSetHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_get_active_subscription_set(realm.handle._pointer)));
+  }
+
+  void refreshSubscriptions(SubscriptionSet subscriptions) {
+    _realmLib.invokeGetBool(() => _realmLib.realm_sync_subscription_set_refresh(subscriptions.handle._pointer));
+  }
+
+  int getSubscriptionSetSize(SubscriptionSet subscriptions) {
+    return _realmLib.realm_sync_subscription_set_size(subscriptions.handle._pointer);
+  }
+
+  SubscriptionHandle subscriptionAt(SubscriptionSet subscriptions, int index) {
+    return SubscriptionHandle._(_realmLib.invokeGetPointer(() => _realmLib.realm_sync_subscription_at(
+          subscriptions.handle._pointer,
+          index,
+        )));
+  }
+
+  SubscriptionHandle? findSubscriptionByName(SubscriptionSet subscriptions, String name) {
+    return using((arena) {
+      return _realmLib
+          .realm_sync_find_subscription_by_name(
+            subscriptions.handle._pointer,
+            name.toUtf8Ptr(arena),
+          )
+          .convert(SubscriptionHandle._);
```

Comment:
How does that work with the convert extension method? Wouldn't `realm_sync_find_subscription_by_name` return `nullptr` instead of `null` - are those handled the same way in the `if (self == null)` check? I don't expect them to be.

Reply:
Problem is we should return null, if `realm_sync_find_subscription_by_name` returns `nullptr`. This handles that gracefully

Reply:
I prefer abstracting repetitive code. I'll basically by copy pasting this pattern all over:
```
final result = someAction();
return result == null ? null : Thing(result);
```
Anyway, I will do that.